### DISCUSSION TITLE
Adding a Custom Dropdown Action for Tags in Rich Text Editor - Error with Controls Array Format

### DISCUSSION THREAD

**Discussion by BSCE21038** [mantinedev/mantine#7107] [Q&A]
Hi,

I'm implementing a custom dropdown action in the Rich Text Editor for inserting specific tags. My goal is to add a button to the toolbar that opens a dropdown menu with predefined tags, which, when selected, insert into the editor content.

However, I’m encountering an error: group.filter is not a function. This suggests that the controls array passed to the Rich Text Editor might be causing an issue due to its format. It seems the component expects each item in the controls array to be a string or an array of strings (for toolbar groups), rather than React components or JSX elements like <IconButton />.

Could you advise on the best way to add a custom dropdown control in the Rich Text Editor toolbar? I’d appreciate any guidance on the expected format for controls or any examples you could share for adding dropdowns with custom actions.

Thanks in advance!

**Comment by rtivital** [comment]
You can find an example for custom control here – https://mantine.dev/x/tiptap/#custom-controls
